wandb/apis/public.py

```diff
@@ -2649,14 +2649,22 @@ def download(self, root=".", replace=False):
         """Downloads a file previously saved by a run from the wandb server.
 
         Arguments:
+            root (str): Local directory or file path to save the file. Defaults to ".".
             replace (boolean): If `True`, download will overwrite a local file
                 if it exists. Defaults to `False`.
-            root (str): Local directory to save the file.  Defaults to ".".
 
         Raises:
             `ValueError` if file already exists and replace=False
         """
-        path = os.path.join(root, self.name)
+        path_ext = os.path.splitext(self.name)[1]
+        if path_ext == "":
+            raise Exception(f"File {self.name} must contain a file extension, otherwise it won't be downloaded!")
+        if root == "/":
+            raise Exception("'/' is not a valid directory, to use the root directory use '.' instead.")
+        root_ext = os.path.splitext(root)[1]
+        if root_ext not in [path_ext, ""]:
+            raise Exception(f"Remote file extension ({path_ext}) doesn't match target file extension ({root_ext})!")
```

Comment:
Ok then the `upload` and `download` functions won't support file renames, just upload to directories, right? If that's the case let me know what do you want to keep from this PR, otherwise I'll close it. Thanks!